### PR TITLE
docs(forms): Add documentation and live examples for NgForm, NgFormCo…

### PR DESCRIPTION
**Pull request by alxhub**
…ntrol, NgFormModel, and NgModel.

### CODE REVIEW

File: modules/angular2/src/core/forms/directives/ng_form.ts [hunk 1]
```diff
@@ -19,36 +19,64 @@ const formDirectiveBinding =
...
+ * or {@link ControlArray}s. This hierarchy is reflected in the form's `value`, a
+ * JSON object that mirrors the form structure.
+ *
+ * # Submission
+ *
+ * The `ng-submit` event signals when the user triggers a form submission.
+ *
+ * ### Example ([live demo](http://plnkr.co/edit/ltdgYj4P0iY64AR71EpL?p=preview))
  *
  *  ```
```
Comment (vicb) [inline modules/angular2/src/core/forms/directives/ng_form.ts]: question: should be specify the language (ie "typescript" after the quotes) ?

Comment (alxhub) [inline modules/angular2/src/core/forms/directives/ng_form.ts]: I don't think there's any harm in adding it, though I don't see it used much in the Angular codebase. There are a few examples, though, so I'll follow suit.

File: modules/angular2/src/core/forms/directives/ng_form_control.ts [hunk 2]
```diff
@@ -12,29 +12,47 @@ const formControlBinding =
     CONST_EXPR(new Binding(NgControl, {toAlias: forwardRef(() => NgFormControl)}));
 
 /**
- * Binds an existing control to a DOM element.
+ * Binds an existing {@link Control} to a DOM element.
  *
- * # Example
+ * Ordinarily, DOM elements that are form controls have a {@link Control}
```
Comment (vsavkin) [inline modules/angular2/src/core/forms/directives/ng_form_control.ts]: Not sure if this is 100% correct. For instance, if you use ng-form-model, then DOM elements will their controls from that model. No new controls will be created automatically.

Comment (alxhub) [inline modules/angular2/src/core/forms/directives/ng_form_control.ts]: Got it, will reword.

Comment (alxhub) [inline modules/angular2/src/core/forms/directives/ng_form_control.ts]: Fixed.

**Pull request closed without merging**